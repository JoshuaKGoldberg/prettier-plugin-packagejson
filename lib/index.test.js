const { describe, it } = require('@jest/globals')
const prettier = require('prettier')

const uglyJson = `{
  "license": "",
  "version":    "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
`

describe('inferred parser', () => {
  it.each(['package.json', 'foo/bar/package.json'])('should format: %s', async filepath => {
    await expect(
      Promise.resolve(
        prettier.format(uglyJson, {
          filepath,
          plugins: ['.'],
        }),
      ),
    ).resolves.toMatchSnapshot()
  })

  it.each([
    'Package.json',
    'package.JSON',
    'package-lock.json',
    'composer.json',
    'package.json/composer.json',
    'foo.json',
  ])('should not format: %s', async filepath => {
    await expect(
      Promise.resolve(
        prettier.format(uglyJson, {
          filepath,
          plugins: ['.'],
        }),
      ),
    ).resolves.toMatchSnapshot()
  })
})

describe('json-stringify parser', () => {
  it.each(['package.json', 'foo/bar/package.json'])('should format: %s', async filepath => {
    await expect(
      Promise.resolve(
        prettier.format(uglyJson, {
          filepath,
          parser: 'json-stringify',
          plugins: ['.'],
        }),
      ),
    ).resolves.toMatchSnapshot()
  })

  it.each([
    'Package.json',
    'package.JSON',
    'package-lock.json',
    'composer.json',
    'package.json/composer.json',
    'foo.json',
    'bar.js',
    undefined,
  ])('should not format: %s', async filepath => {
    await expect(
      Promise.resolve(
        prettier.format(uglyJson, {
          filepath,
          parser: 'json-stringify',
          plugins: ['.'],
        }),
      ),
    ).resolves.toMatchSnapshot()
  })
})
