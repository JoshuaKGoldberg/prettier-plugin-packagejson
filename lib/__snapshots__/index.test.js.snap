// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser: json-stringify { format: false, filepath: 'Package.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: json-stringify { format: false, filepath: 'bar.js' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: json-stringify { format: false, filepath: 'composer.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: json-stringify { format: false, filepath: 'foo.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: json-stringify { format: false, filepath: 'package.JSON' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: json-stringify { format: false, filepath: 'package.json/composer.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: json-stringify { format: false, filepath: 'package-lock.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: json-stringify { format: false, filepath: undefined } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: json-stringify { format: true, filepath: 'foo/bar/package.json' } 1`] = `
"{
  "name": "",
  "version": "",
  "description": "",
  "keywords": [
    "C",
    "B",
    "A"
  ],
  "license": "",
  "author": "",
  "scripts": {
    "prebuild": "",
    "build": "",
    "postbuild": "",
    "dev": "",
    "lint": "",
    "start": "",
    "pretest": "",
    "test": "",
    "posttest": ""
  },
  "dependencies": {
    "A": "*",
    "B": "*"
  },
  "devDependencies": {
    "A": "*",
    "B": "*"
  },
  "peerDependencies": {
    "A": "*",
    "B": "*"
  },
  "optionalDependencies": {
    "A": "*",
    "B": "*"
  }
}
"
`;

exports[`parser: json-stringify { format: true, filepath: 'package.json' } 1`] = `
"{
  "name": "",
  "version": "",
  "description": "",
  "keywords": [
    "C",
    "B",
    "A"
  ],
  "license": "",
  "author": "",
  "scripts": {
    "prebuild": "",
    "build": "",
    "postbuild": "",
    "dev": "",
    "lint": "",
    "start": "",
    "pretest": "",
    "test": "",
    "posttest": ""
  },
  "dependencies": {
    "A": "*",
    "B": "*"
  },
  "devDependencies": {
    "A": "*",
    "B": "*"
  },
  "peerDependencies": {
    "A": "*",
    "B": "*"
  },
  "optionalDependencies": {
    "A": "*",
    "B": "*"
  }
}
"
`;

exports[`parser: undefined { format: false, filepath: 'Package.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: undefined { format: false, filepath: 'composer.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: undefined { format: false, filepath: 'foo.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": ["C", "B", "A"]
}
"
`;

exports[`parser: undefined { format: false, filepath: 'package.JSON' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: undefined { format: false, filepath: 'package.json/composer.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: undefined { format: false, filepath: 'package-lock.json' } 1`] = `
"{
  "license": "",
  "version": "",
  "devDependencies": {
    "B": "*",
    "A": "*"
  },
  "description": "",
  "dependencies": {
    "B": "*",
    "A": "*"
  },
  "scripts": {
    "posttest": "",
    "test": "",
    "start": "",
    "pretest": "",
    "build": "",
    "postbuild": "",
    "prebuild": "",
    "lint": "",
    "dev": ""
  },
  "peerDependencies": {
    "B": "*",
    "A": "*"
  },
  "author": "",
  "name": "",
  "optionalDependencies": {
    "B": "*",
    "A": "*"
  },
  "keywords": [
    "C",
    "B",
    "A"
  ]
}
"
`;

exports[`parser: undefined { format: true, filepath: 'foo/bar/package.json' } 1`] = `
"{
  "name": "",
  "version": "",
  "description": "",
  "keywords": [
    "C",
    "B",
    "A"
  ],
  "license": "",
  "author": "",
  "scripts": {
    "prebuild": "",
    "build": "",
    "postbuild": "",
    "dev": "",
    "lint": "",
    "start": "",
    "pretest": "",
    "test": "",
    "posttest": ""
  },
  "dependencies": {
    "A": "*",
    "B": "*"
  },
  "devDependencies": {
    "A": "*",
    "B": "*"
  },
  "peerDependencies": {
    "A": "*",
    "B": "*"
  },
  "optionalDependencies": {
    "A": "*",
    "B": "*"
  }
}
"
`;

exports[`parser: undefined { format: true, filepath: 'package.json' } 1`] = `
"{
  "name": "",
  "version": "",
  "description": "",
  "keywords": [
    "C",
    "B",
    "A"
  ],
  "license": "",
  "author": "",
  "scripts": {
    "prebuild": "",
    "build": "",
    "postbuild": "",
    "dev": "",
    "lint": "",
    "start": "",
    "pretest": "",
    "test": "",
    "posttest": ""
  },
  "dependencies": {
    "A": "*",
    "B": "*"
  },
  "devDependencies": {
    "A": "*",
    "B": "*"
  },
  "peerDependencies": {
    "A": "*",
    "B": "*"
  },
  "optionalDependencies": {
    "A": "*",
    "B": "*"
  }
}
"
`;
